tests/subsys/pm/power_mgmt/src/dummy_driver.c

```diff
@@ -9,33 +9,88 @@
 #include <pm/device_runtime.h>
 #include "dummy_driver.h"
 
+static bool dont_sleep;
+extern bool z_sys_post_kernel;
+
 static int dummy_open(const struct device *dev)
 {
-	return pm_device_get(dev);
+	dont_sleep = false;
+	pm_device_get(dev);
+	if (dev->pm->enable == false) {
+		pm_device_enable(dev);
+		return pm_device_get(dev);
+
+	}
+	return 0;
 }
 
 static int dummy_close(const struct device *dev)
 {
-	return pm_device_put(dev);
+	dont_sleep = false;
+	pm_device_put(dev);
+	if (z_sys_post_kernel == false) {
+		z_sys_post_kernel = true;
+		return 0;
+	}
+	pm_device_disable(dev);
+	return 0;
+}
```

Comment:
Close should just put the device. What is being done here is not the intention of this API.

Reply:
yes, updated.